lib/date_time/date_time.c

```diff
@@ -27,6 +27,8 @@
 
 LOG_MODULE_REGISTER(date_time, CONFIG_DATE_TIME_LOG_LEVEL);
 
+#define DATE_TIME_UPDATE_INTERVAL_THRESHOLD_MS 700
```

Comment:
I had to chance this to 700ms from 100ms due to NTP which took like 300ms in my experiments. I made a bit of buffer for it.
The real interval is anyway thought to be hours or at least tens of minutes so this shouldn't matter.
@simensrostad Please notice this change.

Reply:
Change noticed. During poor network conditions, getting a response from the various NTP servers can fail / take some time. Based on that should we make `DATE_TIME_UPDATE_INTERVAL_THRESHOLD_MS` be a product of number of fallback servers and `CONFIG_DATE_TIME_NTP_QUERY_TIME_SECONDS`. I mean if we have 3 servers were the response time out on the default `CONFIG_DATE_TIME_NTP_QUERY_TIME_SECONDS` of 5 seconds and the fourth succeeds 700 milliseconds would not be enough?

Reply:
Good comment. This made me think that the whole 700ms or 100ms or 20s is just far too much of guess work. Should we just start the timer when we have received an updated time. Then the next time is received slightly later (same 100ms, 700ms or even 20s depending on method and network conditions) than `CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS`. I wouldn't expect this inaccuracy to be a problem.

Reply:
Sounds like a better approach to me.

Reply:
Sounds good!

Reply:
This change has been done and cleans up the code nicely by getting rid of `DATE_TIME_UPDATE_INTERVAL_THRESHOLD_MS`.